### PR TITLE
make class backward compatible

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

**Comment by teclator** [comment]
So with this PR we fix the already known problem with old lense described in #66

**Comment by jreidinger** [comment]
@teclator yes

File: src/lib/cfa/ntp_conf.rb [hunk 1]
```diff
@@ -394,10 +394,54 @@ class RestrictRecord < Record
 
       def options
         return [] unless tree_value?
-        augeas_options.map { |option| option[:value] }
+        res = augeas_options.map { |option| option[:value] }
+        # backward compatibility with old lense that set value ip restriction
```
Comment (mvidner) [inline src/lib/cfa/ntp_conf.rb]: lens

File: src/lib/cfa/ntp_conf.rb [hunk 2]
```diff
@@ -394,10 +394,54 @@ class RestrictRecord < Record
...
+        #     action[2]: nofail
+        #
+        #   line in configuration file:
+        #     restrict default nofail
+        #   augeas tree:
+        #     key:       restrict
+        #     value:     default
+        #     action[1]: nofail
+        #
+        # with new lense value is always address like:
```
Comment (mvidner) [inline src/lib/cfa/ntp_conf.rb]: lens

File: src/lib/cfa/ntp_conf.rb [hunk 3]
```diff
@@ -394,10 +394,54 @@ class RestrictRecord < Record
...
+        #     key: restrict
+        #     value: default
+        #     action[1]: nofail
+        res.shift if ["-6", "-4"].include?(value)
+
+        res
       end
 
       def options=(options)
+        # backward compatibility with old lense that set value ip restriction
```
Comment (mvidner) [inline src/lib/cfa/ntp_conf.rb]: lens

**Comment by teclator** [comment]
@jreidinger thnx for make it backward compatible

**Pull request merged**